File: check-if-array-is-sorted-and-rotated.py
class Solution(object):
    def check(self, nums):
        """
        :type nums: List[int]
        :rtype: bool
        """
        n=len(nums)
        cnt=0

        for i in range(1,n):
            if nums[i-1]>nums[i]:
                cnt+=1

        if nums[0]<nums[n-1]:
            cnt+=1

        return cnt<=1

File: check-if-array-is-sorted-and-rotated.java
class Solution {
    public boolean check(int[] nums) {
        int n = nums.length;

        int drops = 0;

        for(int i = 0; i < n; i++) {
            if(nums[i] > nums[(i + 1) % n]) {
                drops++;
            }
        }

        return drops <= 1;
    }
}
